txcache/selectionTracker.go

```diff
@@ -0,0 +1,147 @@
+package txcache
+
+import (
+	"bytes"
+	"math/big"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+)
+
+type accountBreadcrumb struct {
+	initialNonce    uint64
+	lastNonce       uint64
+	consumedBalance *big.Int
+}
+
+type trackedBlock struct {
+	nonce                uint64
+	hash                 []byte
+	rootHash             []byte
+	prevHash             []byte
+	breadcrumbsByAddress map[string]*accountBreadcrumb
+}
+
+type selectionTracker struct {
+	mutState          sync.RWMutex
+	mutLatestRootHash sync.RWMutex
+	latestNonce       uint64
+	latestRootHash    []byte
+	state             []*trackedBlock
+	txCache           *TxCache
+}
+
+func newTrackedBlock(nonce uint64, blockHash []byte, rootHash []byte, prevHash []byte) *trackedBlock {
+	return &trackedBlock{
+		nonce:                nonce,
+		hash:                 blockHash,
+		rootHash:             rootHash,
+		prevHash:             prevHash,
+		breadcrumbsByAddress: make(map[string]*accountBreadcrumb),
+	}
+}
+
+// NewSelectionTracker - creates a new selectionTracker
+func NewSelectionTracker(txCache *TxCache) (*selectionTracker, error) {
+	if txCache == nil {
```

Comment:
Even not strictly necessary now, perhaps use `check.IfNil()`, in case we'll use an interface in the future here?